universe/Universe.cpp

```diff
@@ -2726,8 +2721,8 @@ std::set<int> Universe::RecursiveDestroy(int object_id) {
         Destroy(object_id);
         retval.insert(object_id);
 
-    } else if (auto fleet = std::dynamic_pointer_cast<Fleet>(obj)) {
-        for (int ship_id : fleet->ShipIDs()) {
+    } else if (auto obj_fleet = std::dynamic_pointer_cast<Fleet>(obj)) {
```

Comment:
why?

Reply:
warning for shadowed on line 2762 `for (auto& fleet : m_objects.FindObjects<Fleet>())`